From 803f4cb978d92359b78f087ac96eca85c0058abf Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 28 Aug 2024 10:27:33 +0300
Subject: [PATCH] [FSW-8694] fs1: get requested manifest data

Complete manifest data required to validate its
key, signature. Hence, support required to get
manifest data address after SPI DMA of requested
index. Manifest memory can be accessed using boot
pointer structure to retrive boot pointer members

Change-Id: I0aecc490ee74e8873fb35849d3c5ad9d2ea150bc
---

diff --git a/common/boot_desc.c b/common/boot_desc.c
index dd620bc..4dbc517 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -259,6 +259,14 @@
 	return !!(p_manifest_ctx->boot_desc_flag);
 }
 
+uint32_t boot_desc_get_boot_pointer(enum hl_fw_manifest_type manifest_type)
+{
+	if (!boot_rom_boot_desc_init_done(manifest_type))
+		return 0;
+
+	return p_spi_manifest_const->dest_addr_arc;
+}
+
 static bool boot_rom_is_valid_start_addr(struct boot_desc *desc)
 {
 	if ((desc->start_addr >= desc->loadaddr) &&
diff --git a/common/boot_desc.h b/common/boot_desc.h
index ef414e3..1bc5f13 100644
--- a/common/boot_desc.h
+++ b/common/boot_desc.h
@@ -37,6 +37,7 @@
 						enum boot_rom_progress img_sts_base,
 						enum hl_fw_img_type img);
 uint32_t boot_desc_get_recovery_img_count(enum hl_fw_manifest_type manifest_idx);
+uint32_t boot_desc_get_boot_pointer(enum hl_fw_manifest_type manifest_type);
 
 #endif /* _BOOT_DESC_H_ */
 
